model/project_parser.go

```diff
@@ -587,7 +587,7 @@ func LoadProjectInto(ctx context.Context, data []byte, opts *GetProjectOpts, ide
 		}
 		// read from the patch diff if a change has been made in any of the include files
 		if opts.ReadFileFrom == ReadFromPatch || opts.ReadFileFrom == ReadFromPatchDiff {
-			if opts.PatchOpts.patch != nil && opts.PatchOpts.patch.ConfigChanged(path.FileName) {
+			if opts.PatchOpts.patch != nil && opts.PatchOpts.patch.ShouldPatchFileWithDiff(path.FileName) {
```

Comment:
I tested this manually in staging using evergreen-ci/commit-queue-sandbox#332 (which failed to create a patch before I pushed the changes to staging and succeeded after); did you mean something besides that?

Reply:
The manual test looks fine to me. I meant an automated unit test.

Reply:
This specific path is difficult to unit test because it relies on getting a file from Github, both in the success and failure cases.

Reply:
This specific path is difficult to unit test because it relies on getting a file from Github, both in the success and failure cases.  Specifically https://github.com/evergreen-ci/evergreen/blob/362c9a78a9435cd26107d15acd0177de1861d3bf/model/project_parser.go#L671 is called both for ReadFromPatch and ReadFromPatchDiff and would fail in either case, but the step we need to test is after that (i.e. that we don't reach 681). I can test that ReadFileInto is set correctly however.

Reply:
Actually I'm going to refactor to make this more unit testable.